src/services/item/folderService.ts

```diff
@@ -0,0 +1,32 @@
+import { UUID } from '@graasp/sdk';
+
+import { Repositories } from '../../utils/repositories';
+import { Actor } from '../member/entities/member';
+import { FolderItem } from './entities/Item';
+import { ItemServiceManager } from './itemServiceManager';
+import { ItemService } from './service';
+
+export class FolderService extends ItemService {
+  async copy(actor: Actor, repositories: Repositories, itemId: UUID, args: { parentId?: UUID }) {
+    const item = (await this.get(actor, repositories, itemId)) as FolderItem;
+
+    const descendants = await repositories.itemRepository.getDescendants(item);
+    const items = [...descendants, item];
+
+    const res = await super.copy(actor, repositories, itemId, args);
```

Comment:
Maybe it is clearer to do the parent's copy first, and then defining the const concerning the subclass's copy logic.